web-admin-dashboard/src/components/dashboard/sidebar/Sidebar.tsx

```diff
@@ -1,25 +1,31 @@
-import { BsGrid1X2Fill, BsPeopleFill, BsFillGearFill } from "react-icons/bs";
-import "../../../pages/DashboardPage.css";
-import "./Sidebar.css";
-import logo from "../../../assets/dashboard/logo.svg";
+import React, { useState } from 'react';
+import { BsGrid1X2Fill, BsPeopleFill, BsFillGearFill } from 'react-icons/bs';
+import { BiLogOut } from 'react-icons/bi';
+import '../../../pages/DashboardPage.css';
+import './Sidebar.css';
+import logo from '../../../assets/dashboard/logo.svg';
+import { Link, useResolvedPath, useMatch } from 'react-router-dom';
+import { ReactNode } from 'react';
+import LogoutDialog from './LogoutDialog'; // Import the LogoutDialog component
 
 interface Props {
   openSidebar: () => void;
   openSidebarToggle: boolean;
-  handleClickDashboard: (event: React.MouseEvent) => void;
-  handleClickUser: (event: React.MouseEvent) => void;
 }
-function Sidebar({
-  openSidebarToggle,
-  openSidebar,
-  handleClickDashboard,
-  handleClickUser,
-}: Props) {
+
+function Sidebar({ openSidebarToggle, openSidebar }: Props) {
+  const [isLogoutDialogOpen, setIsLogoutDialogOpen] = useState(false);
+
+  const openLogoutDialog = () => {
```

Comment:
We can talk about a way how to make this cleaner, currently just trying to get code out.